tests/eko/test_kernels_ns.py

```diff
@@ -117,34 +117,93 @@ def test_ode_nlo():
             np.testing.assert_allclose(lhs, rhs, atol=np.abs(delta_a))
 
 
-#  def test_ode_nnlo():
-#  nf = 3
-#  ev_op_iterations = 10
-#  gamma_ns = np.random.rand(3) + 0j
-#  delta_a = -1e-6
-#  a0 = 0.3
-#  for a1 in [0.1, 0.2]:
-#  r = (a1 * gamma_ns[0] + a1 ** 2 * gamma_ns[1] + a1 ** 3 * gamma_ns[2]) / (
-#  beta.beta_qcd((0,0), nf) * a1 ** 2
-#  + beta.beta_qcd((1,0), nf) * a1 ** 3
-#  + beta.beta_qcd((1,0), nf) * a1 ** 4
-#  )
-#  for method in ["iterate-exact"]:
-#  rhs = r * ns.dispatcher(2, method, gamma_ns, a1, a0, nf, ev_op_iterations)
-#  lhs = (
-#  ns.dispatcher(
-#  2, method, gamma_ns, a1 + 0.5 * delta_a, a0, nf, ev_op_iterations
-#  )
-#  - ns.dispatcher(
-#  2, method, gamma_ns, a1 - 0.5 * delta_a, a0, nf, ev_op_iterations
-#  )
-#  ) / delta_a
-#  np.testing.assert_allclose(lhs, rhs, atol=np.abs(delta_a))
+def test_ode_nnlo():
```

Comment:
Tests generate now a
```
tests/eko/test_kernels_ns.py::test_gamma_usage
  /home/felix/Physik/N3PDF/EKO/eko/src/eko/kernels/non_singlet.py:300: RuntimeWarning: invalid value encountered in cdouble_scalars
    e *= e0 * num / den
```
which sounds potentially dangerous - can we explain this (and suppress) or avoid?

Reply:
I don't see any warning when running the tests: 

```
(eko_dev) giacomomagni@Giacomos-MacBook-Pro eko % poe test              
Poe => pytest tests
========================================================================================= test session starts ==========================================================================================
platform darwin -- Python 3.10.4, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
rootdir: /Volumes/Git_Workspace/physicstools/NN3PDF/eko, configfile: pyproject.toml
plugins: env-0.6.2, pyfakefs-4.5.6, cov-3.0.0
collected 214 items                                                                                                                                                                                    

tests/eko/test_ad.py ...                                                                                                                                                                         [  1%]
tests/eko/test_ad_aem1.py ...                                                                                                                                                                    [  2%]
tests/eko/test_ad_aem2.py ...                                                                                                                                                                    [  4%]
tests/eko/test_ad_as1.py ......                                                                                                                                                                  [  7%]
tests/eko/test_ad_as1aem1.py ....                                                                                                                                                                [  8%]
tests/eko/test_ad_as2.py .                                                                                                                                                                       [  9%]
tests/eko/test_ad_as3.py .                                                                                                                                                                       [  9%]
tests/eko/test_ad_as4.py .....                                                                                                                                                                   [ 12%]
tests/eko/test_basis_rotation.py ...                                                                                                                                                             [ 13%]
tests/eko/test_beta.py ........                                                                                                                                                                  [ 17%]
tests/eko/test_compatibility.py ..                                                                                                                                                               [ 18%]
tests/eko/test_couplings.py ..........                                                                                                                                                           [ 22%]
tests/eko/test_ev_op_flavors.py ........                                                                                                                                                         [ 26%]
tests/eko/test_ev_op_grid.py .....                                                                                                                                                               [ 28%]
tests/eko/test_ev_op_physical.py .....                                                                                                                                                           [ 31%]
tests/eko/test_ev_operator.py ......                                                                                                                                                             [ 34%]
tests/eko/test_f_functions.py ..........                                                                                                                                                         [ 38%]
tests/eko/test_g_functions.py .........                                                                                                                                                          [ 42%]
tests/eko/test_gamma.py .                                                                                                                                                                        [ 43%]
tests/eko/test_harmonics.py ......                                                                                                                                                               [ 46%]
tests/eko/test_init.py .                                                                                                                                                                         [ 46%]
tests/eko/test_interpolation.py ...............                                                                                                                                                  [ 53%]
tests/eko/test_kernels_as4_ei.py .....                                                                                                                                                           [ 56%]
tests/eko/test_kernels_ei.py ......                                                                                                                                                              [ 58%]
tests/eko/test_kernels_ns.py .......                                                                                                                                                             [ 62%]
tests/eko/test_kernels_s.py ........                                                                                                                                                             [ 65%]
tests/eko/test_kernels_utils.py .                                                                                                                                                                [ 66%]
tests/eko/test_matching.py .                                                                                                                                                                     [ 66%]
tests/eko/test_matching_n3lo.py ...                                                                                                                                                              [ 68%]
tests/eko/test_matching_nlo.py ...                                                                                                                                                               [ 69%]
tests/eko/test_matching_nnlo.py ......                                                                                                                                                           [ 72%]
tests/eko/test_mellin.py ....                                                                                                                                                                    [ 74%]
tests/eko/test_member.py .........                                                                                                                                                               [ 78%]
tests/eko/test_msbar_masses.py ...                                                                                                                                                               [ 79%]
tests/eko/test_ome.py .......                                                                                                                                                                    [ 83%]
tests/eko/test_output.py .......                                                                                                                                                                 [ 86%]
tests/eko/test_polygamma.py ..                                                                                                                                                                   [ 87%]
tests/eko/test_runner.py ....                                                                                                                                                                    [ 89%]
tests/eko/test_sv_expanded.py ....                                                                                                                                                               [ 91%]
tests/eko/test_sv_exponentiated.py ..                                                                                                                                                            [ 92%]
tests/eko/test_thresholds.py ...............                                                                                                                                                     [ 99%]
tests/ekomark/test_apply.py ..                                                                                                                                                                   [100%]

---------- coverage: platform darwin, python 3.10.4-final-0 ----------
Coverage HTML written to dir htmlcov
Coverage XML written to file coverage.xml


========================================================================================= 214 passed in 30.58s =========================================================================================
```

Reply:
Mmm ... I get
```
$ poe test -k test_kernels
Poe => pytest tests -k test_kernels
========================================= test session starts ==========================================
platform linux -- Python 3.10.5, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
rootdir: /home/felix/Physik/N3PDF/EKO/eko, configfile: pyproject.toml
plugins: cov-3.0.0, env-0.6.2
collected 214 items / 187 deselected / 27 selected                                                     

tests/eko/test_kernels_as4_ei.py .....                                                           [ 18%]
tests/eko/test_kernels_ei.py ......                                                              [ 40%]
tests/eko/test_kernels_ns.py .......                                                             [ 66%]
tests/eko/test_kernels_s.py ........                                                             [ 96%]
tests/eko/test_kernels_utils.py .                                                                [100%]

=========================================== warnings summary ===========================================
tests/eko/test_kernels_ns.py::test_gamma_usage
  /home/felix/Physik/N3PDF/EKO/eko/src/eko/kernels/non_singlet.py:300: RuntimeWarning: invalid value encountered in cdouble_scalars
    e *= e0 * num / den

tests/eko/test_kernels_s.py::test_gamma_usage
  /home/felix/Physik/N3PDF/EKO/eko/src/eko/anomalous_dimensions/__init__.py:69: RuntimeWarning: invalid value encountered in cdouble_scalars
    c = 1.0 / det

-- Docs: https://docs.pytest.org/en/stable/warnings.html

---------- coverage: platform linux, python 3.10.5-final-0 -----------
Coverage HTML written to dir htmlcov
Coverage XML written to file coverage.xml

============================ 27 passed, 187 deselected, 2 warnings in 5.81s ============================
```
I run a different Python then you but that shouldn't matter I assume ... and the warning is also there on github https://github.com/N3PDF/eko/runs/7282141412?check_suite_focus=true ... I see in your env there is still `pyfakefs` although this is no longer in the `pyproject.toml` - did you run out of sync with the env?

Reply:
I see, well I'll fix them..
 
`pyfakefs`  is coming from here? https://github.com/N3PDF/eko/blob/744e2f9444ae1fdef6676b31483e040eda4403ad/poetry.lock#L435